pkg/cache/tas_flavor_snapshot.go

```diff
@@ -540,16 +550,38 @@ func (s *TASFlavorSnapshot) findTopologyAssignment(
 	podSetTolerations := info.Tolerations
 	podSetNodeSelectors := info.NodeSelector
 	count := tasPodSetRequests.Count
+
+	// If slice topology is not requested then we can assume that slice is a single pod
+	sliceSize, reason := getSliceSizeWithSinglePodAsDefault(tasPodSetRequests.PodSet.TopologyRequest)
```

Comment:
Can we consider moving initializing all arguments used afterwards in the algorithm to a separate function? I'm thinking L542-L593

Reply:
I think that's a good idea. Since it won't simplify reviewing this PR. I'm adding that to a follow-up PR, not to delay this PR.